src/middleware/error-handler.ts

```diff
@@ -0,0 +1,60 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCode } from "status-code-enum";
+
+export class RouterError {
+    statusCode: number;
+    message: string;
+    // NOTE: eslint is required because the goal of RouterError.data is to "return any necessary data" - mostly used for debugging purposes
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data?: any | undefined;
+    catchErrorMessage?: string | undefined;
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(statusCode?: number, message?: string, data?: any, catchErrorMessage?: string) {
+        if (statusCode) {
+            this.statusCode = statusCode;
+        } else {
+            this.statusCode = 500;
+        }
```

Comment:
Can we one-liner these?

```suggestion
this.statusCode = statusCode ?? 500;
```